### PR TITLE
feat(common): CHP-7101 use @commitlint/read to for commits

### PR DESCRIPTION
**Pull request by chanceaclark**
## What/why?

Reads commit messages from the `@commitlint/read` package instead of manually spawning child processes.

### CODE REVIEW

File: bin/validate-commits.js [hunk 1]
```diff
@@ -39,29 +39,42 @@ if (argv.installGitHooks) {
...
+      from,
+      to,
+    };
+  }
+
+  if (process.env.TRAVIS_PULL_REQUEST && process.env.TRAVIS_PULL_REQUEST !== 'false') {
+    return {
+      from: process.env.TRAVIS_BRANCH,
+      to: process.env.TRAVIS_BRANCH,
+    };
```
[the remark this answers is not in the document]
Comment (icatalina) [reply bin/validate-commits.js:74]: This is how we got the list of commits in travis. I'm not sure if any of the repos using `validate-commits` are still in travis, might be worth checking... 🤷

File: bin/validate-commits.js [hunk 2]
```diff
@@ -39,29 +39,42 @@ if (argv.installGitHooks) {
   process.exit(0);
 }
 
-const range = argv._[0] || `${getBranch()}..`;
+read(getBranch())
```
Comment (icatalina) [inline bin/validate-commits.js:42]: 🍹 Why not put everything in a function and use `async` for everything?

```js
async function run() {
  const commits = await read(getBranch());
  const results = commitValidator.validate(commits);
 
  reporter.printReport(results);

  if (!results.valid && !argv.warning) {
    throw new Error('Commit format error!');
  }
}

run()
  .catch((error) => {
    log(error);
    process.exit(2);
  }); 
```

**Review by icatalina — APPROVED** [review]
It looks good, just a style comment 👍

File: bin/validate-commits.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (chanceaclark) [reply bin/validate-commits.js:74]: I searched github for repos, and so far there was about 5 (some archived). Deprecating this Travis supporting code might be a good idea, but definitely for another PR. We should be using CircleCI anyways 🤷

File: bin/validate-commits.js (hunk 2 above)
Comment (chanceaclark) [reply bin/validate-commits.js:42]: Why not.

**Pull request merged**